cerberus/tests/test_assorted.py

```diff
@@ -27,3 +31,23 @@ def test_that_test_fails(test, document):
         pass
     else:
         raise AssertionError("test didn't fail")
+
+
+def test_dynamic_types():
+    decimal_type = TypeDefinition('decimal', (Decimal,), ())
+    document = {'measurement': Decimal(0)}
+    schema = {'measurement': {'type': 'decimal'}}
+
+    validator = Validator()
+    validator.types_mapping['decimal'] = decimal_type
+    # REMOVE
+    validator.rules['type']['allowed'] += ('decimal',)
+    assert_success(document, schema, validator)
+
+    class MyValidator(Validator):
+        types_mapping = Validator.types_mapping.copy()
```

Comment:
alternatively, the metaclass could do this and then update the copy with the `types_mapping` defined for the subclass. but this would make it impossible to populate that mapping from scratch.